### PR TITLE
Delete legacy decryption code, authenticate connection msgs

### PR DESCRIPTION
**Pull request by Patrik-Stas**
Signed-off-by: Patrik Stas <patrik.stas@absa.africa>

### CODE REVIEW

File: libvcx/src/aries/handlers/connection/agent_info.rs [hunk 1]
```diff
@@ -80,60 +90,64 @@ impl AgentInfo {
...
     }
 
-    pub fn decode_message(&self, message: &Message) -> VcxResult<A2AMessage> {
+    fn decode_message(&self, message: &Message, expected_sender_vk: &str) -> VcxResult<A2AMessage> {
         trace!("Agent::decode_message >>> message = {:?}", json!(&message).to_string());
+        EncryptionEnvelope::auth_unpack(message.payload()?, &expected_sender_vk)
+    }
 
-        EncryptionEnvelope::open(message.payload()?)
+    fn decode_message_noauth(&self, message: &Message) -> VcxResult<A2AMessage> {
```
Comment (mirgee) [inline libvcx/src/aries/handlers/connection/agent_info.rs:148]: Maybe `decrypt_decode_message_noauth` is more accurate

**Review by mirgee — CHANGES_REQUESTED** [review]

File: libvcx/src/api/utils.rs [hunk 2]
```diff
@@ -287,6 +287,8 @@ pub extern fn vcx_download_agent_messages(command_handle: u32,
 /// #Returns
 /// Error code as a u32
 #[no_mangle]
+#[deprecated(since = "0.12.0", note = "This is dangerous because downloaded messages are not \
+authenticated and a message appearing to be received from certain connection might been spoofed.")]
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply libvcx/src/api/utils.rs:291]: Okay

File: libvcx/src/aries/mod.rs [hunk 3]
```diff
@@ -634,6 +666,7 @@ pub mod test {
     }
 
     #[cfg(feature = "pool_tests")]
+    #[cfg(feature = "to_restore")] // todo: this is the problem alice.download_message("credential-offer");
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply libvcx/src/aries/mod.rs:669]: good catch, restored

File: libvcx/src/aries/utils/encryption_envelope.rs [hunk 4]
```diff
@@ -160,6 +196,6 @@ pub mod tests {
             _ => return assert!(false)
         };
 
-        assert_eq!(ack, EncryptionEnvelope::open(message_2).unwrap());
+        assert_eq!(ack, EncryptionEnvelope::anon_unpack(message_2).unwrap());
```
[the remark this answers is not in the document]
Comment (Patrik-Stas) [reply libvcx/src/aries/utils/encryption_envelope.rs:207]: tests added

**Comment by Patrik-Stas** [comment]
@mirgee great points, addressed, ready for re-review

**Review by mirgee — APPROVED** [review]

**Pull request merged**